src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
+    private int size;
+    private int cursor;
+
+
+    private String errorMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
```

Comment:
Strings are immutable, so every time you use the concatenation operator (`+`), you copy the content of both strings to create another. Prefer `StringBuilder` over concantenation. However, I suggest using `String.format()` here, cause the number of strings here won't affect performance that much, but this way code looks neat and log-like.
```suggestion
    private String errorMsg(int index) {
        return String.format("Index: %d, Size: %d", index, size);
    }
```